tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_section_order.py

```diff
@@ -0,0 +1,220 @@
+
+    BODY = 1
+    HEADER = 2
+    BODY_AND_HEADER = 3
+
+
+def get_expected_code_exception(
+    section_kind, section_test, test_position
+) -> tuple[str, EOFException | None]:
+    """
+    Verification vectors with code and exception based on test combinations
+    """
+    match (section_kind, section_test, test_position):
+        case (SectionKind.TYPE, SectionTest.MISSING, CasePosition.HEADER):
+            return "ef000102000100030400010000800001305000ef", EOFException.MISSING_TYPE_HEADER
+        case (SectionKind.TYPE, SectionTest.MISSING, CasePosition.BODY):
+            return (
+                "ef0001010004020001000304000100305000ef",
+                EOFException.INVALID_SECTION_BODIES_SIZE,
+            )
+        case (SectionKind.TYPE, SectionTest.MISSING, CasePosition.BODY_AND_HEADER):
+            return "ef0001020001000304000100305000ef", EOFException.MISSING_TYPE_HEADER
+        case (SectionKind.TYPE, SectionTest.WRONG_ORDER, CasePosition.HEADER):
+            return (
+                "ef000102000100030100040400010000800001305000ef",
+                EOFException.MISSING_TYPE_HEADER,
+            )
+        case (SectionKind.TYPE, SectionTest.WRONG_ORDER, CasePosition.BODY):
+            return (
+                "ef000101000402000100030400010030500000800001ef",
+                # TODO why invalid first section type? it should say that the body incorrect
+                EOFException.INVALID_FIRST_SECTION_TYPE,
+            )
+        case (SectionKind.TYPE, SectionTest.WRONG_ORDER, CasePosition.BODY_AND_HEADER):
+            return (
+                "ef000102000100030100040400010030500000800001ef",
+                EOFException.MISSING_TYPE_HEADER,
+            )
+        case (SectionKind.CODE, SectionTest.MISSING, CasePosition.HEADER):
+            return "ef00010100040400010000800001305000ef", EOFException.MISSING_CODE_HEADER
+        case (SectionKind.CODE, SectionTest.MISSING, CasePosition.BODY):
+            return (
+                "ef000101000402000100030400010000800001ef",
+                # TODO should be an exception of empty code bytes, because it can understand that
+                # last byte is data section byte
+                EOFException.INVALID_SECTION_BODIES_SIZE,
+            )
+        case (SectionKind.CODE, SectionTest.MISSING, CasePosition.BODY_AND_HEADER):
+            return "ef00010100040400010000800001ef", EOFException.MISSING_CODE_HEADER
+        case (SectionKind.CODE, SectionTest.WRONG_ORDER, CasePosition.HEADER):
+            return (
+                "ef000101000404000102000100030000800001305000ef",
+                EOFException.MISSING_CODE_HEADER,
+            )
+        case (SectionKind.CODE, SectionTest.WRONG_ORDER, CasePosition.BODY):
+            return (
+                "ef000101000402000100030400010000800001ef305000",
+                EOFException.UNDEFINED_INSTRUCTION,
+            )
+        case (SectionKind.CODE, SectionTest.WRONG_ORDER, CasePosition.BODY_AND_HEADER):
```

Comment:
With this test we can remove "data_section_preceding_code_section" from `container.py`.

Reply:
shall I remove it in this PR?

Reply:
hm, I actually like this redundancy. the container.py looks more like a UnitTest for python implementation.